magda-sleuther-format-layerer/src/format-engine/measures/dcatFormatMeasure.ts

```diff
@@ -0,0 +1,116 @@
+/*
+* Tries to determine the format by deciphering DCAT-DISTRIBUTION-STRING -> format
+* TODO Unit tests
+*/
+
+import { Record } from "@magda/typescript-common/dist/generated/registry/api";
+import { getCommonFormat } from "../formats";
+import MeasureResult from "./MeasureResult";
+
+function applyTransform(format: string, transformFormat: any) {
+    const newFormat = transformFormat(format);
+    return newFormat;
+}
+
+function lowerCase(str: string) {
+    return str.toLowerCase();
+}
+
+function removePeriod(str: string) {
+    return str.split(".").join("");
+}
+
+function replaceCommaThenSpace(str: string) {
+    return str.split(", ").join(" ");
+}
+
+function replaceSpaceThenComma(str: string) {
+    return str.split(" ,").join(" ");
+}
+
+function replaceComma(str: string) {
+    return str.split(",").join(" ");
+}
+
+export default function getMeasureResult(
+    relatedDistribution: Record
+): MeasureResult {
+    const { format } = relatedDistribution.aspects["dcat-distribution-strings"];
+
+    if (format === null || format === "") {
+        return null;
+    }
+
+    let processedFormat: string;
+
+    const transforms = [
+        (str: string) =>
+            lowerCase(
+                removePeriod(
+                    replaceComma(
+                        replaceSpaceThenComma(replaceCommaThenSpace(str))
+                    )
+                )
+            )
+    ];
+
+    for (const tf of transforms) {
+        processedFormat = applyTransform(format, tf);
+    }
+
+    // hard coded rules for separating out multiple formats when provided
+    let splitFormat: Array<string> = [];
+    let finalFormat: Array<string> = [];
+
+    // e.g: application/exe
+    splitFormat = processedFormat.split("/");
```

Comment:
dcat-distribution-strings actually has a `mediaType` field which the mime type is supposed to go in, might be worth looking in that too.